oak-run/src/test/java/org/apache/jackrabbit/oak/plugins/document/CreateGarbageCommandTest.java

```diff
@@ -0,0 +1,437 @@
+            assertTrue(garbageRootNodeChildNames.contains(generateBasePath));
+        }
+
+        String generateBasePath = generatedBasePaths.get(0);
+        for (int i = 0; i < garbageNodesParentCount; i ++) {
+
+            for (int j = 0; j < nodesPerParent; j++) {
+
+                // check deleted node paths
+                List<String> deletedNodePaths = CreateGarbageCommand.generateGapOrphanNodePaths(
+                        generateBasePath, i, j, orphansDepth, orphansLevelGap, false);
+                for (String deletedNodePath : deletedNodePaths) {
+                    NodeDocument missingDocument = getDocument(cmd, Collection.NODES, deletedNodePath, 0);
+                    assertNull(missingDocument);
+                }
+
+                // check not deleted node paths - remaining gap orphans
+                List<String> notDeletedNodePaths = CreateGarbageCommand.generateGapOrphanNodePaths(generateBasePath, i, j, orphansDepth, orphansLevelGap, true);
+                for (String notDeletedNodePath : notDeletedNodePaths) {
+                    NodeDocument gapOrphanDocument = getDocument(cmd, Collection.NODES, notDeletedNodePath, 0);
+                    assertNotNull(gapOrphanDocument);
+                }
+            }
+        }
+    }
+
+    /**
+     * Utility method that returns the root node of the full gc generated garbage nodes.
+     * @param nodeStore
+     * @return
+     */
+    private static @NotNull NodeState getGarbageRootNode(DocumentNodeStore nodeStore) {
+        NodeState garbageRootNodeState = nodeStore.getRoot().getChildNode(CreateGarbageCommand.GARBAGE_GEN_ROOT_PATH_BASE).
+                getChildNode(CreateGarbageCommand.GARBAGE_GEN_ROOT_NODE_NAME);
+        return garbageRootNodeState;
+    }
+
+    /**
+     * Used in unit tests for retrieving a document by path from the document store using the documentNodeStore
+     * of this class.
+     * @param collection
+     * @param key
+     * @param maxCacheAge
+     * @return
+     * @param <T>
+     * @throws DocumentStoreException
+     */
+    @Nullable
+    private  <T extends Document> T getDocument(CreateGarbageCmd cmd, Collection<T> collection, String key, int maxCacheAge)
+            throws DocumentStoreException {
+        return cmd.getCommand().getDocumentNodeStore().getDocumentStore().find(collection, key, maxCacheAge);
+    }
+
+    /**
+     * Wrapper class for the GenerateGarbageCommand to be used in unit tests.
+     */
+    private static class CreateGarbageCmd implements Runnable {
+
+        private final ImmutableList<String> args;
+        private CreateGarbageCommand command;
+        private Closer closer;
+        private List<String> generatedBasePaths;
+
+        public CreateGarbageCmd(String... args) {
+            // append the default mongodb connection string if one was not provided
+            if (args[0].startsWith("mongodb://")) {
+                this.args = ImmutableList.copyOf(args);
```

Comment:
please avoid the usage of guava collections. We have this OAK-10685 epic to remove guava from Oak.